File: resources/rosenbrock-1.py
def rosenbrock(x,p,nargout):
    """
    Generalized n-dim rosenbrock function with steepness parameter p[0]
    """
    scale=p[0]
    n=len(x)
    f=0
    if nargout>1:
        g=np.zeros((n,1))
    for k in range(n-1):
        T=x[k+1,0]-x[k,0]**2
        S=1-x[k,0]
        f+=scale*T**2+S**2
        if nargout>1:
            g[k,0]=-4*scale*x[k,0]*T-2*S
            if k>0:
                g[k,0]+=2*scale*(x[k,0]-x[k-1,0]**2)
    if nargout>1:
        g[n-1,0]=2*scale*(x[n-1,0]-x[n-2,0]**2)
        return f,g
    else:
        return f
    

File: resources/rosenbrock-1.m
function [f,g]=rosenbrock(x,a)

n=length(x);
f=0;
g=zeros(n,1);
for k=1:n-1
    T=x(k+1)-x(k)^2;
    S=1-x(k);
    f=f+a*T^2+S^2;
    g(k)=-4*a*x(k)*T-2*S;
    if k>1
        g(k)=g(k)+2*a*(x(k)-x(k-1)^2);
    end
end

g(n)=2*a*(x(n)-x(n-1)^2);

return